### ISSUE TITLE
STEP 2.10.0 Extended (Finalize) gets stuck

### ISSUE THREAD

**Issue by SomeDargons** [airbreather/StepperUpper#43]
After running through core, core finalize, and extended, the extended finalize gets stuck at 22 tasks remaining. Left it at that part for about 6 hours and no change. Started the whole thing over again and it got stuck at 22 tasks again

**Comment by airbreather** [comment]
What's the FileVersion say at the top of all your .xml files?

**Comment by SomeDargons** [comment]
1.0.0.8
I originally just edited the ones from the step forum with the changes in #40, but once I saw that it'd been merged, I downloaded them from github and tried again.

If it's relevant, I'm trying to have Stepper Upper set up everything on my C drive (an SSD), but the mods and the program itself are kept on my second drive (HDD).

**Comment by airbreather** [comment]
Odd... counting it up from the bottom, 22 tasks remaining means that it got stuck before the DSR patch generation process could complete... I'm not sure how that would have happened.

If you can't get this working, then you may need to manually go through the "Finalize" process... it would involve running through just Core + Extended, then manually do the stuff I listed on #19, skipping steps 4, 24, and 34,

**Comment by SomeDargons** [comment]
Just tried it again (with Stepper on the SSD), specifically watching to see what it did and it opens Mod Organizer momentarily, but nothing comes of it, no other programs start running.

I'll go through and do the Extended Finalize process manually.

**Comment by SomeDargons** [comment]
I figured out what it was, DSR was taking exception to my java installations (despite having both 32 and 64 versions up to date); I wasn't able to run it manually either, so I reinstalled java and it started working. Going to retry Stepper with the Extended Finalize.

**Comment by SomeDargons** [comment]
It worked. Extended Finalize went through without a problem. Thank you for pointing me towards the manual steps.

In case this happens to other people, consider putting the manual steps into the wiki, or having the program show/describe the current (and/or past) task(s) that it's working on.

**Comment by airbreather** [comment]
This is some good evidence that #20 is a useful thing to do.

> In case this happens to other people, consider putting the manual steps into the wiki, [...]

It's something to think about, but the main reason why I keep "Thing" / "Thing_Finalize" separated these days is to let an experienced user use the StepperUpper process as a **starting point** for their own stuff without having to sit through those long things twice.

> [...] or having the program show/describe the current (and/or past) task(s) that it's working on.

That sounds difficult to pull off in a way that looks "nice", especially without adding more GUI stuff to this (which I want to continue to avoid).

Another way you could have seen what was going on would have been by right-clicking the "H" in your notification area / system tray... IIRC, AutoHotkey has a way to see what script it's running and where it currently is within that script.

Short version... it sucks that this happened, I'm glad I could help you figure it out, and I'm reconsidering my prioritization of #20, but I don't think there's anything more I can / should do here.
